### PR TITLE
Pebble implementation added.

### PR DESCRIPTION
**Pull request by berkantay**
I have added LevelDB first but we found that it is not actively maintained anymore. So I switched to Pebble DB inspired from LevelDB. Thanks for the review 👓

### CODE REVIEW

File: .github/release-drafter-leveldb.yml
Comment (ReneWerner87) [inline .github/release-drafter-leveldb.yml:1]: pls rename filename

File: pebble/README.md
Comment (ReneWerner87) [inline pebble/README.md:1]: pls add description

**Review by ReneWerner87 — CHANGES_REQUESTED** [review]
- [x] pls add a part in the root README.md

File: pebble/README.md
Comment (berkantay) [reply pebble/README.md:1]: Description added ✔️

File: .github/release-drafter-leveldb.yml
Comment (berkantay) [reply .github/release-drafter-leveldb.yml:1]: filename renamed ✔️

**Comment by berkantay** [comment]
Change requests applied.

File: README.md [hunk 1]
```diff
@@ -48,45 +48,47 @@ type Storage interface {
...
-* [SQLite3](/sqlite3) <a href="https://github.com/gofiber/storage/actions?query=workflow%3A%22Tests+Sqlite3%22">
-    <img src="https://img.shields.io/github/actions/workflow/status/gofiber/storage/test-sqlite3.yml?branch=main&label=%F0%9F%A7%AA%20&style=flat&color=75C46B">
+- [SQLite3](/sqlite3) <a href="https://github.com/gofiber/storage/actions?query=workflow%3A%22Tests+Sqlite3%22">
+  <img src="https://img.shields.io/github/actions/workflow/status/gofiber/storage/test-sqlite3.yml?branch=main&label=%F0%9F%A7%AA%20&style=flat&color=75C46B">
   </a>
-* [S3](/s3) <a href="https://github.com/gofiber/storage/actions?query=workflow%3A%22Tests+S3%22">
-    <img src="https://img.shields.io/github/actions/workflow/status/gofiber/storage/test-s3.yml?branch=main&label=%F0%9F%A7%AA%20&style=flat&color=75C46B">
+- [S3](/s3) <a href="https://github.com/gofiber/storage/actions?query=workflow%3A%22Tests+S3%22">
+  <img src="https://img.shields.io/github/actions/workflow/status/gofiber/storage/test-s3.yml?branch=main&label=%F0%9F%A7%AA%20&style=flat&color=75C46B">
+- [Pebble](/pebble) <a href="https://github.com/gofiber/storage/actions?query=workflow%3A%22Tests+Pebble%22">
```
Comment (ReneWerner87) [inline README.md:92]: sort it alphabetically

File: README.md [hunk 2]
```diff
@@ -48,45 +48,47 @@ type Storage interface {
 
 ## 📑 Storage Implementations
 
-* [ArangoDB](/arangodb) <a href="https://github.com/gofiber/storage/actions?query=workflow%3A%22Tests+ArangoDB%22">
```
Comment (ReneWerner87) [inline README.md:51]: image
why did you changed the control character ?

**Review by ReneWerner87 — CHANGES_REQUESTED** [review]
pls fix the last one and then we are ready

File: README.md (hunk 1 above)
Comment (berkantay) [reply README.md]: alphabetically sorted 👍

File: .github/workflows/test-pebble.yml [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+  pull_request:
+    paths:
+      - "pebble/**"
+name: "Tests pebble"
+jobs:
+  Tests:
+    strategy:
+      matrix:
+        go-version:
+          - 1.18.x
```
Comment (gaby) [inline .github/workflows/test-pebble.yml:17]: The go.mod uses 1.19, remove 1.18.x from here

File: pebble/README.md [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+func (s *Storage) Reset() error
+func (s *Storage) Close() error
+func (s *Storage) Conn() *badger.DB
+```
+
+### Installation
+
+Pebble is tested on the 2 last [Go versions](https://golang.org/dl/) with support for modules. So make sure to initialize one first if you didn't do that yet:
+
+```bash
```
Comment (gaby) [inline pebble/README.md:29]: Add a small comment here saying this is only if you dont have an existing module

File: pebble/pebble.go [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+	if err != nil {
+		return nil, err
+	}
+
+	if data == nil {
+		return nil, nil
+	}
+
+	if err := closer.Close(); err != nil {
+		log.Fatal(err)
```
Comment (gaby) [inline pebble/pebble.go:59]: Not a fan of this, but other storage drivers are doing the same. They should probably be fix in a separate PR

File: pebble/go.mod [hunk 6]
```diff
@@ -0,0 +1,33 @@
+module github.com/gofiber/storage/pebble
```
Comment (gaby) [inline pebble/go.mod:1]: Just in case run `go mod tidy` to cleanup anything from the go.sum file

File: pebble/internal/validator.go [hunk 7]
```diff
@@ -0,0 +1,21 @@
+package internal
```
Comment (gaby) [inline pebble/internal/validator.go:1]: Do we really need a full file for 1 function?

**Review by gaby — CHANGES_REQUESTED** [review]

**Comment by gaby** [comment]
@berkantay Yeah go 1.18 needs to be removed, PebbleDB only supports 1.19/1.20

File: pebble/pebble.go (hunk 5 above)
Comment (berkantay) [reply pebble/pebble.go:59]: In this case I did follow the other storage drivers. It could be changed accordingly.

File: pebble/internal/validator.go (hunk 7 above)
Comment (berkantay) [reply pebble/internal/validator.go:1]: I put that just to keep things separated. But I can also move it into `pebble.go`.

**Review by ReneWerner87 — APPROVED** [review]

**Comment by ReneWerner87** [comment]
i think only the last change is needed in the README

**Comment by ReneWerner87** [comment]
dont see a adjustment in the readme for gaby´s comment

**Review by gaby — APPROVED** [review]
LGTM

**Review by efectn — APPROVED** [review]

**Comment by ReneWerner87** [comment]
@berkantay it's okay if we change that
https://github.com/gofiber/storage/blob/main/pebble/pebble.go#LL129C32-L129C36
from 0644 to 0600 ?
because the security check has noted that

**Comment by ReneWerner87** [comment]
https://github.com/gofiber/storage/actions/runs/5276258446/jobs/9542732742
image

can you provide fixes for these hints ?
